src/STEMMUS_SCOPE_exe.m

```diff
@@ -1,11 +1,82 @@
 %%%%%%% A function to run STEMMUS_SCOPE %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-function STEMMUS_SCOPE_exe(config_file)
+function STEMMUS_SCOPE_exe(config_file, runMode)
+    if ~exist('runMode', 'var')
+        runMode = "full";
+    end
 
     % disable all warnings
     warning('off', 'all');
 
-    % set the global variable CFG
+    % set the variable CFG
     CFG = config_file;
+    bmiMode = "none";
 
-    % run STEMMUS_SCOPE main code
-    run STEMMUS_SCOPE;
+    % If the runMode is "full" or was not provided, the model will run as normal
+    if strcmp(runMode, "full")
+        run STEMMUS_SCOPE;
+
+    elseif strcmp(runMode, "bmi")
+        % In interactive mode MATLAB stays open and waits for a new command...
+
+        % Define BMI required variable names:
+        bmiVarNames = {'ModelSettings', ... % Model settings struct
+                       'TimeStep', ... % Time step size (in seconds)
+                       'KT', ... % Index of current time step
+                       'SiteProperties', ... % Site properties (e.g. lat, lon)
+                       'fluxes', ... % Atmospheric fluxes
+                       'TT' ... % Soil temperature over depth
+                      }; %#ok
+
+        isInitialized = false;
+
+        % ...until finalize has been run, at which point it quits.
+        while ~strcmp(bmiMode, "finalize")
+            bmiMode = input("\nFinished command. Select run mode: ", "s");
+
+            if startsWith(bmiMode, "initialize ")
+                % Get config file:

```

Comment:
`eraseBetween` is not octave compatible. Also, with start 1 and end 11, the config file should be inserted without quotes and spaces. Please see my suggestion below.

Reply:
Thanks for the code suggestion!

The BMI still won't work with Octave due to the Matlab 7.3 format not being supported. I don't expect that to be ever implemented in Octave as it's been an issue for more than a decade, even though an open MATLAB file IO library has been available for years https://github.com/tbeu/matio